src/main/java/com/um/inventoryservice/BusinessLayer/EmployeeServiceImpl.java

```diff
@@ -42,6 +42,7 @@ public Mono<EmployeeDTO> updateEmployee(String employeeId, Mono<EmployeeDTO> emp
                 .flatMap(e -> employeeDTOMono
                         .map(EntityDTOUtil::toEntity)
                         .doOnNext(x -> x.setEmployeeId(e.getEmployeeId()))
+                        .doOnNext(x -> x.setId(e.getId()))
```

Comment:
Can a comment just be left here on why this line is important and what it does, for future reference?

Reply:
Setting the id ensures the id is taken from the employee that is being updated that's saved in the database for a successful and correct update: if the line is removed, another id will be set and a duplicate entity will be created and added rather than updating the existing one.